staff_features/accessions/step_definitions/accession_edit.rb

```diff
@@ -0,0 +1,188 @@
+# frozen_string_literal: true
+
+Given 'the Accession appears in the search results list' do
+  visit "#{STAFF_URL}/accessions"
+
+  fill_in 'filter-text', with: @uuid
+
+  within '.search-filter' do
+    find('button').click
+  end
+
+  search_result_rows = all('#tabledSearchResults tbody tr')
+  expect(search_result_rows.length).to eq 1
+
+  within search_result_rows[0] do
+    element = find('a', text: 'Edit')
+
+    @accession_id = URI.decode_www_form_component(element[:href]).split('/').pop
+  end
+end
+
+Given 'the Accession is opened in the view mode' do
+  visit "#{STAFF_URL}/accessions"
+
+  fill_in 'filter-text', with: @uuid
+
+  within '.search-filter' do
+    find('button').click
+  end
+
+  search_result_rows = all('#tabledSearchResults tbody tr')
+  expect(search_result_rows.length).to eq 1
+
+  within search_result_rows[0] do
+    element = find('a', text: 'View')
+
+    @accession_id = URI.decode_www_form_component(element[:href]).split('/').pop
+  end
+
+  click_on 'View'
+end
+
+Given 'the Accession is opened in edit mode' do
+  visit "#{STAFF_URL}/accessions"
+
+  fill_in 'filter-text', with: @uuid
+
+  within '.search-filter' do
+    find('button').click
+  end
+
+  search_result_rows = all('#tabledSearchResults tbody tr')
+  expect(search_result_rows.length).to eq 1
+
+  within search_result_rows[0] do
+    element = find('a', text: 'Edit')
+
+    @accession_id = URI.decode_www_form_component(element[:href]).split('/').pop
+  end
+
+  click_on 'Edit'
+end
+
+When 'the user changes the {string} field to {string}' do |field, value|
+  fill_in field, with: value
+end
+
+When 'the user changes the {string} field' do |field|
+  fill_in field, with: "Updated Title #{SecureRandom.uuid}"
+end
+
+Then 'the {string} field has the original value' do |field|
+  expect(page).to have_field(field, with: "Accession Title #{@uuid}")
+end
+
+Then 'the Accession is updated with the new {string} as {string}' do |field, value|
+  fill_in field, with: value
+end
+
+Then 'the Accession is opened in the edit mode' do
+  uri = current_url.split('/')
+  uri.pop
+  accession_id = uri.pop
+
+  expect(accession_id).to eq @accession_id
+end
+
+Then 'all fields become editable' do
+  fill_in 'Identifier', with: 'Identifier'
+end
+
+Then 'the field {string} has value {string}' do |field, value|
+  expect(page).to have_field(field, with: value)
+end
+
+Then 'the Accession Date field has the original value' do
+  visit "#{STAFF_URL}/accessions/#{@accession_id}/edit"
+
+  expect(page).to have_field('Accession Date', with: '2000-01-01')
```

Comment:
minor point but it might be better to have the original values as constants in case we decide to change them for some reason.

Reply:
Done.

However, I believe that maybe there should be a consistency between the original values, since some of them use strings interpolated with variables (uuids, or maybe something else), and it's not clear to the step definition writer when to use what.

The writer also checks the original values from the form that sees in the screen, so it is more straightforward to directly copy the strings, rather than having an extra lookup to another file, which may grow quite a lot in the future. In addition to that, in the future, we may have multiple records/factories, which may have multiple original values that may differ slightly, which again can cause confusion.

We can always refactor this this if it causes issues in the future.

Reply:
Ok I see that you added ORIGINAL_ACCESSION_DATE as a constant, so can you change line 91 to use it in the test?

Reply:
Done.